plugins/out_stackdriver/stackdriver_operation.c

```diff
@@ -0,0 +1,164 @@
+                field_name = flb_sds_create_len(p->key.via.str.ptr, p->key.via.str.size);
+                
+                if (strcmp(field_name, OPERATION_FIELD_IN_JSON) == 0 && p->val.type == MSGPACK_OBJECT_MAP) {
+                    flb_sds_destroy(*operation_id);
+                    flb_sds_destroy(*operation_producer);
+
+                    *operation_id = flb_sds_create("");
+                    *operation_producer = flb_sds_create("");
+                    *operation_first = false;
+                    *operation_last = false;
+
+                    op_status = OPERATION_EXISTED;
+
+                    msgpack_object sub_field = p->val;
+                    if (sub_field.via.map.size != 0) {
+                        msgpack_object_kv* tmp_p = sub_field.via.map.ptr;
+                        msgpack_object_kv* const tmp_pend = sub_field.via.map.ptr + sub_field.via.map.size;
+
+                        for (; tmp_p < tmp_pend; ++tmp_p) {
+                            if (tmp_p->key.type != MSGPACK_OBJECT_STR) {
+                                op_status = EXTRA_OR_INVALID_TYPE;
+                                break;
+                            }
+                            else {   
+                                flb_sds_destroy(sub_field_name);
+                                sub_field_name = flb_sds_create_len(tmp_p->key.via.str.ptr, tmp_p->key.via.str.size);
+
+                                if (strcmp(sub_field_name, "id") == 0 && tmp_p->val.type == MSGPACK_OBJECT_STR) {
+                                    flb_sds_destroy(*operation_id);
+                                    *operation_id = flb_sds_create_len(tmp_p->val.via.str.ptr, tmp_p->val.via.str.size);
+                                }
+                                else if (strcmp(sub_field_name, "producer") == 0 && tmp_p->val.type == MSGPACK_OBJECT_STR) {
+                                    flb_sds_destroy(*operation_producer);
+                                    *operation_producer = flb_sds_create_len(tmp_p->val.via.str.ptr, tmp_p->val.via.str.size);
+                                }
+                                else if (strcmp(sub_field_name, "first") == 0 && tmp_p->val.type == MSGPACK_OBJECT_BOOLEAN) {
+                                    *operation_first = tmp_p->val.via.boolean;
+                                }
+                                else if (strcmp(sub_field_name, "last") == 0 && tmp_p->val.type == MSGPACK_OBJECT_BOOLEAN) {
+                                    *operation_last = tmp_p->val.via.boolean;
+                                }
+                                else {
+                                    /* extra sub-fiels or incorrect type of sub-fields */
+                                    op_status = EXTRA_OR_INVALID_TYPE;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                    
+                    break;
```

Comment:
[optional to address] I think this break is too far away from the `if` condition, so I suggest move this break to the beginning of the if condition and reverse the condition.

Reply:
Done